code/game/machinery/dna_scanner.dm

```diff
@@ -159,9 +159,10 @@
 			return
 
 		var/mob/M = G.affecting
-		M.loc = loc
-		user.stop_pulling()
+		M.forceMove(loc)
```

Comment:
So the mob's vision is properly updated so if someone adds a special vision when inside that machine it will work correctly.